src/panels/config/script/ha-script-editor.ts

```diff
@@ -550,6 +573,30 @@ export class HaScriptEditor extends KeyboardShortcutMixin(LitElement) {
     }
   }
 
+  private async _duplicate() {
+    if (this._dirty) {
+      if (
+        !(await showConfirmationDialog(this, {
+          text: this.hass!.localize(
+            "ui.panel.config.common.editor.confirm_unsaved"
+          ),
+          confirmText: this.hass!.localize("ui.common.yes"),
+          dismissText: this.hass!.localize("ui.common.no"),
+        }))
+      ) {
+        return;
+      }
+      // Wait for dialog to complate closing
+      await new Promise((resolve) => setTimeout(resolve, 0));
+    }
+    showScriptEditor(this, {
+      ...this._config,
+      alias: `${this._config?.alias} (${this.hass.localize(
+        "ui.panel.config.automation.picker.duplicate"
```

Comment:
This should still be script, but in the `en.json` you can reference the translation of the automation section.
```suggestion
        "ui.panel.config.script.picker.duplicate"
```

Reply:
Oh, that's cool! Let me try again.

Reply:
Does that look right?